### PR TITLE
Add docs for for storing resource state format

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

Add docs for for storing resource state format

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

### CODE REVIEW

File: docs/docsite/rst/stored_state_schema.rst [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+Specification format for storing resource state
+===============================================
+
+Storing resource state allows users to have a detailed view about the resources they have provisioned earlier.
+By storing resource state, a user would be able to reference their defined resources and declare a state of `absent` and expect Ansible to resolve the relationships between resources and determine the order of operations rather than requiring the user to write a properly ordered playbook.
+This resource state specification is provider-agnostic and allows some resource group metadata to be stored in a `State` dictionary.
+
+The format for storing resource state should be something like:
+
+```yaml
```
Comment (gravesm) [inline docs/docsite/rst/stored_state_schema.rst:12]:
```suggestion
```json
```

File: docs/docsite/rst/stored_state_schema.rst [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+              },
+              "Modified": {
+                  "description":"Date and time in RFC3339 format that the ResourceGroup was last modified.",
+                  "type":"date"
+              },
+              "ResourceGroup": {
+                  "description":"The unique identifier for a group of resources.",
+                  "type":"string"
+              },
+              <anything else we think makes sense>
```
Comment (gravesm) [inline docs/docsite/rst/stored_state_schema.rst:35]: I think we can just remove this line.

File: docs/docsite/rst/stored_state_schema.rst [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+          }
+        },
+        "Resources":{
+            "description":"Dictionary of resources belonging to ResourceGroupName.",
+            "type":"object",
+            "properties": {
```
Comment (gravesm) [inline docs/docsite/rst/stored_state_schema.rst:41]: Same as for the resource declaration format, I think this should use `patternProperties`.

File: docs/docsite/rst/stored_state_schema.rst [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    }]
+
+```
+
+Example of Storing Resource State
+=================================
+
+For the sake of simplicity, an example of storing resource state definition for AWS provider might look something like this:
+
+```yaml
```
Comment (gravesm) [inline docs/docsite/rst/stored_state_schema.rst:63]:
```suggestion
```json
```

**Comment by alinabuzachis** [comment]
recheck

**Review by gravesm — APPROVED** [review]

**Comment by gravesm** [comment]
regate

**Pull request merged**